src/draw/lv_draw_vector.h

```diff
@@ -0,0 +1,213 @@
+
+/*Param operations*/
+void lv_vpath_dsc_init(lv_vpath_dsc_t * dsc);
+void lv_vpath_set_scissor(lv_vpath_dsc_t * dsc, lv_area_t * area);
+void lv_vpath_set_path_transform(lv_vpath_dsc_t * dsc, lv_matrix_t * matrix);
+void lv_vpath_set_fill_color(lv_vpath_dsc_t * dsc, lv_color_t color);
+void lv_vpath_set_fill_opa(lv_vpath_dsc_t * dsc, lv_opa_t opa);
+void lv_vpath_set_fill_rule(lv_vpath_dsc_t * dsc, lv_vpath_fill_t rule);
+void lv_vpath_set_fill_pattern_mode(lv_vpath_dsc_t * dsc, lv_vpath_pattern_t mode);
+void lv_vpath_set_fill_image(lv_vpath_dsc_t * dsc, lv_draw_img_dsc_t * img_dsc);
+void lv_vpath_set_fill_grad(lv_vpath_dsc_t * dsc, lv_grad_dsc_t * grad);
+void lv_vpath_set_fill_transform(lv_vpath_dsc_t * dsc, lv_matrix_t * matrix);
+void lv_vpath_set_stroke_color(lv_vpath_dsc_t * dsc, lv_color_t color);
+void lv_vpath_set_stroke_opa(lv_vpath_dsc_t * dsc, lv_opa_t opa);
+void lv_vpath_set_stroke_width(lv_vpath_dsc_t * dsc, float width);
+/**
+ * @param[in] dash_pattern The array of consecutive pair values of the dash length and the gap length.
+ * @param[in] dash_pattern_count The length of the @p dash_pattern array.
+ */
+void lv_vpath_set_stroke_dash(lv_vpath_dsc_t * dsc, float * dash_pattern, uint16_t dash_pattern_count);
+/**
+ * @param[in] cap The cap style value. The default value is @c LV_VPATH_STROKE_CAP_SQUARE.
+ */
+void lv_vpath_set_stroke_cap(lv_vpath_dsc_t * dsc, lv_vpath_stroke_cap_t cap);
+/**
+ * The join style is used for joining the two line segment while stroking the path.
+ * @param[in] join The join style value. The default value is @c LV_VPATH_STROKE_JOIN_BEVEL.
+ */
+void lv_vpath_set_stroke_join(lv_vpath_dsc_t * dsc, lv_vpath_stroke_join_t join);
+/**
+ * @param[in] miterlimit The miterlimit imposes a limit on the extent of the stroke join, when the @c LV_VPATH_STROKE_JOIN_MITER is set. The default value is 4.
+ */
+void lv_vpath_set_stroke_miter_limit(lv_vpath_dsc_t * dsc, uint16_t miter_limit);
+void lv_vpath_set_stroke_grad(lv_vpath_dsc_t * dsc, lv_grad_dsc_t * grad);
+void lv_vpath_set_order(lv_vpath_dsc_t * dsc, bool stroke_first);
+
+/*Draw path commands*/
+lv_vpath_task_t * lv_vpath_task_create(void);
+void lv_vpath_draw(struct _lv_layer_t * layer, lv_vpath_task_t * task, lv_vpath_t * path, lv_vpath_dsc_t * dsc);
+void lv_vpath_clear(struct _lv_layer_t * layer, lv_vpath_task_t * task, lv_area_t * rect, lv_color_t color);
+void lv_vpath_submit(struct _lv_layer_t * layer, lv_vpath_task_t * task);
+void lv_vpath_draw_oneshot(struct _lv_layer_t * layer, lv_vpath_t * path, lv_vpath_dsc_t * dsc);
+
+/*Helper functions*/
+void lv_matrix_translate(lv_matrix_t * matrix, float dx, float dy);
+void lv_matrix_scale(lv_matrix_t * matrix, float scale_x, float scale_y);
+void lv_matrix_rotate(lv_matrix_t * matrix, float angle);
+/*Pseudo 3D rotation (using affine transformations to simulate 3d rotations cast to xy plane)*/
+void lv_matrix_rotate_p3d(lv_matrix_t * matrix, float rx, float ry, float rz);
```

Comment:
“
void lv_matrix_rotate_p3d(lv_matrix_t * matrix, float rx, float ry, float rz);
void lv_matrix_rotate_quaternion(lv_matrix_t * matrix, float q0, float q1, float q2, float q3);
”
If you want to implement 3D transformation, you need **matrix[4][4]**, not **matrix[3][3]**